.github/workflows/docs-gen-and-push.yaml

```diff
@@ -34,7 +34,7 @@ jobs:
 
       - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # tag=v5.5.0
         with:
-          go-version: v1.23.10
+          go-version-file: go.mod
```

Comment:
I have similar concerns here, will this simply pick 1.24.0 because that's what is in the `go.mod`? I would prefer to keep this explicit to be honest.